### PR TITLE
Better identity handling

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: executor/src/witgen/generator.rs [hunk 1]
```diff
@@ -122,7 +106,9 @@ where
                     }
                 };
 
-                self.handle_eval_result(result);
+                if self.handle_eval_result(result) {
+                    progress = true;
+                }
```
Comment (Schaeff) [inline executor/src/witgen/generator.rs:111]:
```suggestion
                progress |= self.handle_eval_result(result);
```

File: executor/src/witgen/generator.rs [hunk 2]
```diff
@@ -133,12 +119,14 @@ where
                         && column.query.is_some()
                     {
                         let result = self.process_witness_query(&column);
-                        self.handle_eval_result(result)
+                        if self.handle_eval_result(result) {
+                            progress = true;
+                        }
```
Comment (Schaeff) [inline executor/src/witgen/generator.rs:124]: same

File: executor/src/witgen/generator.rs [hunk 3]
```diff
@@ -204,16 +192,7 @@ where
...
-                IdentityKind::Polynomial => {
-                    self.process_polynomial_identity(identity.left.selector.as_ref().unwrap())
-                }
-                IdentityKind::Plookup | IdentityKind::Permutation => self.process_plookup(identity),
-                kind => {
-                    unimplemented!("Identity of kind {kind:?} is not supported in the executor")
-                }
-            };
-            if result.is_err() {
+            if self.process_identity(identity).is_err() {
```
Comment (Schaeff) [inline executor/src/witgen/generator.rs:195]: The error gets lost here. Is it not needed?

**Review by Schaeff — COMMENTED** [review]
Just a few comments but could you explain the big picture here?

File: executor/src/witgen/generator.rs (hunk 3 above)
Comment (chriseth) [reply executor/src/witgen/generator.rs:193]: This is the loop part. If we fail with the proposed values, we try the regular deduction method that could succeed or at least provide a better error.

**Review by Schaeff — APPROVED** [review]

**Pull request merged**